contracts/modules/grouping/GroupingModule.sol

```diff
@@ -0,0 +1,224 @@
+            if (!LICENSE_REGISTRY.hasIpAttachedLicenseTerms(ipIds[i], groupLicenseTemplate, groupLicenseTermsId)) {
+                revert Errors.GroupingModule__IpHasNoGroupLicenseTerms(
+                    ipIds[i],
+                    groupLicenseTemplate,
+                    groupLicenseTermsId
+                );
+            }
+            pool.addIp(groupIpId, ipIds[i]);
+        }
+
+        emit AddedIpToGroup(groupIpId, ipIds);
+    }
+
+    /// @notice Removes IP from group.
+    /// the function must be called by the Group IP owner or an authorized operator.
+    /// @param groupIpId The address of the group IP.
+    /// @param ipIds The IP IDs.
+    function removeIp(address groupIpId, address[] calldata ipIds) external whenNotPaused verifyPermission(groupIpId) {
+        _checkIfGroupMembersLocked(groupIpId);
+        // remove ip from group
+        GROUP_IP_ASSET_REGISTRY.removeGroupMember(groupIpId, ipIds);
+        for (uint256 i = 0; i < ipIds.length; i++) {
+            IGroupRewardPool(GROUP_IP_ASSET_REGISTRY.getGroupRewardPool(groupIpId)).removeIp(groupIpId, ipIds[i]);
+        }
+        emit RemovedIpFromGroup(groupIpId, ipIds);
+    }
+
+    /// @notice Claims reward.
+    /// @param groupId The address of the group.
+    /// @param token The address of the token.
+    /// @param ipIds The IP IDs.
+    function claimReward(address groupId, address token, address[] calldata ipIds) external whenNotPaused {
+        IGroupRewardPool pool = IGroupRewardPool(GROUP_IP_ASSET_REGISTRY.getGroupRewardPool(groupId));
+        // claim reward from group IPA's RoyaltyVault to group pool
+        pool.collectRoyalties(groupId, token);
+        // trigger group pool to distribute rewards to group members vault
+        uint256[] memory rewards = pool.distributeRewards(groupId, token, ipIds);
+        for (uint256 i = 0; i < ipIds.length; i++) {
+            emit ClaimedReward(groupId, token, ipIds[i], rewards[i]);
+        }
+    }
+
+    function name() external pure override returns (string memory) {
+        return GROUPING_MODULE_KEY;
+    }
+
+    /// @notice Returns the available reward for each IP in the group.
+    /// @param groupId The address of the group.
+    /// @param token The address of the token.
+    /// @param ipIds The IP IDs.
+    /// @return The rewards for each IP.
+    function getClaimableReward(
+        address groupId,
+        address token,
+        address[] calldata ipIds
+    ) external view returns (uint256[] memory) {
+        // get claimable reward from group pool
+        IGroupRewardPool pool = IGroupRewardPool(GROUP_IP_ASSET_REGISTRY.getGroupRewardPool(groupId));
+        return pool.getAvailableReward(groupId, token, ipIds);
+    }
+
+    /// @dev The group members are locked if the group has derivative IPs or license tokens minted.
+    function _checkIfGroupMembersLocked(address groupIpId) internal view {
+        if (LICENSE_REGISTRY.hasDerivativeIps(groupIpId)) {
```

Comment:
Is there any way of having derivatives with `LICENSE_TOKEN.getTotalTokensByLicensor(groupIpId)`?

Reply:
Yes, there is function `registerDerivative` can register derivative to group IP without need group IP minting any licenseToken.